var/spack/repos/builtin/packages/gdb/package.py

```diff
@@ -80,11 +82,20 @@ class Gdb(AutotoolsPackage, GNUMirrorPackage):
     depends_on("source-highlight", when="+source-highlight")
     depends_on("ncurses", when="+tui")
     depends_on("gmp", when="@11.1:")
+    depends_on("elfutils@0.178:+debuginfod", when="@11.1:+debuginfod")
 
     build_directory = "spack-build"
 
     def configure_args(self):
-        args = ["--with-system-gdbinit={0}".format(self.prefix.etc.gdbinit)]
+        args = [
+            "--with-system-gdbinit={0}".format(self.prefix.etc.gdbinit),
+            *self.enable_or_disable("lto"),
+            *self.enable_or_disable("quad"),
```

Comment:
Should this be using `with_or_without()` for `quad`?

Reply:
I can build successfully but am not sure why the `*self...` syntax works.

Reply:
> I can build successfully but am not sure why the *self... syntax works.

`enable_or_disable` returns a list, so this is the same as:
```python
>>> l = [2, 3]
>>> [1, *l, 4]
[1, 2, 3, 4]

```

Reply:
Good catch for `with_or_without('quad')`.  Should be pushed now @tldahlgren

Reply:
> > I can build successfully but am not sure why the *self... syntax works.
> 
> `enable_or_disable` returns a list, so this is the same as:
> 
> ```python
> >>> l = [2, 3]
> >>> [1, *l, 4]
> [1, 2, 3, 4]
> ```

I must've done something wrong when I tried that with a list of strings and with a local method that returned a list of strings.  Both reported syntax errors.